### PR TITLE
repro: add failing test for trace segment net assignment bug

### PR DESCRIPTION
**Pull request by itisrohit**
### Summary of Changes
This pull request introduces a dedicated **reproduction unit test case** in `tests/pcb/repros/repro19-segment-net-assignment-bug.test.tsx` to demonstrate a bug where trace segments lose their net assignment.

**Note**: To comply with "no failing tests on merge", the test asserts the *current failing behavior* (getting assigned to net `0` / unconnected) and contains a detailed comment explaining the bug, ensuring the CI suite remains green.

---

### The Bug Description
In `lib/pcb/stages/AddTracesStage.ts`, when a trace's `source_trace_id` represents a `source_net` ID and `connection_name` is absent, the converter fails to resolve the connectivity key. This causes the generated trace segments to be assigned to net `0` (unconnected) in the KiCad S-expression output.

### CODE REVIEW

**Comment by mohan-bee** [comment]
use real tsx. also checks failing.

**Comment by itisrohit** [comment]
> use real tsx. also checks failing.

done!

File: tests/pcb/repros/repro19-segment-net-assignment-bug.test.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  const sourceNet = circuitJson.find((el) => el.type === "source_net")
+  expect(sourceNet).toBeDefined()
+  const sourceNetId = sourceNet.source_net_id
+
+  // Find the pcb_trace and modify it to simulate the bug
+  const pcbTrace = circuitJson.find((el) => el.type === "pcb_trace")
+  expect(pcbTrace).toBeDefined()
+
+  // Simulate the bug:
+  // 1. Delete connection_name
```
Comment (mohan-bee) [inline tests/pcb/repros/repro19-segment-net-assignment-bug.test.tsx]: i don't understand why are you simulating a bug here... please add a snapshot, not clear !

Comment (itisrohit) [reply tests/pcb/repros/repro19-segment-net-assignment-bug.test.tsx]: > i don't understand why are you simulating a bug here... please add a snapshot, not clear !

I have added a visual PNG snapshot for this test and committed the baseline image to `__snapshots__`.

**Why we modify the circuit JSON in-memory:**
When rendering using standard TSX components, the `tscircuit` React engine automatically outputs a fully valid structure (with `connection_name` and a `source_trace_id` pointing to `source_trace`), which avoids triggering the bug. 

However, in real-world imports or external autorouter outputs (e.g. from `capacity-autorouter`), the generated `pcb_trace` has `source_trace_id` pointing to a `source_net` ID and lacks `connection_name`. 

Modifying the rendered output in-memory allows us to write the test in real TSX while still testing the converter's handling of this specific buggy JSON structure. I have added detailed comments in the code explaining this.

**Review by mohan-bee — CHANGES_REQUESTED** [review]
no you are just modifying the circuit json to simulate a bug, not good.

**Comment by itisrohit** [comment]
> no you are just modifying the circuit json to simulate a bug, not good.

I have updated the test to use a clean, static, declarative `circuitJson` input array, completely removing any dynamic in-memory object mutations. I also added a visual PNG snapshot matching this output.

### Why static JSON is used instead of TSX for this test:
1. **The Bug Conditions**: The bug only manifests when the incoming `pcb_trace` has `source_trace_id` pointing directly to a `source_net` ID and does **not** have a `connection_name` property. (This is a valid Circuit JSON structure produced by third-party tools like `capacity-autorouter` or custom schematic-to-PCB converters).
2. **TSX Compiler Output**: The standard `tscircuit` React/TSX compiler/renderer always automatically generates `connection_name` on all trace elements it renders. There is no way via TSX props to compile a board where a trace lacks `connection_name`.
3. **The Solution**: If we used standard TSX, the test would pass even without the fix applied because the converter would fallback to `connection_name`. To actually replicate this edge case and prevent regression, we must feed the converter a static `circuitJson` array where `connection_name` is absent. This aligns with how similar edge cases (like `repro08` for vias and `repro12` for standalone via sizes) are tested.

**Review by techmannih — COMMENTED** [review]
@itisrohit Both snapshots look the same to me. I'm not seeing any issue.

**Comment by itisrohit** [comment]
@techmannih Yes, visually the PNG/SVG snapshots look identical because the physical geometry of the copper trace is exactly the same in both cases.

However, electrically and in the generated `.kicad_pcb` file output:
- **With the bug**: The trace gets assigned to `(net 0)` (unconnected/no net). This means there is no electrical connectivity mapping in KiCad, which will trigger DRC (Design Rule Check) errors about unconnected nets/pads.
- **Expected behavior**: The trace should be assigned to `(net 1)` (corresponding to `net_a`).

The unit test asserts this data-level discrepancy programmatically:
`expect(firstMatch[1]).toBe("0")` (confirming that under the current buggy code, it gets net `0`).

**Comment by techmannih** [comment]
@itisrohit I am not sure, tag @seveibar @imrishabh18 for review

**Comment by itisrohit** [comment]
cc @seveibar  @imrishabh18 for review

**Review by MustafaMulla29 — CHANGES_REQUESTED** [review]
Cant see any difference in the snapshot, you have to create a better repro

**Comment by itisrohit** [comment]
> Cant see any difference in the snapshot, you have to create a better repro

You are absolutely correct! Net assignment is a data-level / electrical connectivity bug, not a visual layout bug. The copper track geometry is physically identical in both cases, which is why the PNG snapshots looked the same.

To resolve the confusion, I have removed the visual PNG snapshot from the test (similar to how the via net assignment test `repro08` does not have a snapshot). The test now strictly verifies the data-level connectivity issue by checking if the net ID inside the generated `.kicad_pcb` file is correctly resolved.

File: tests/pcb/repros/repro19-segment-net-assignment-bug.test.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    {
+      type: "source_net",
+      source_net_id: "net_a",
+      name: "net_a",
+      subcircuit_connectivity_map_key: "net_a_key",
+    },
+    {
+      type: "pcb_trace",
+      pcb_trace_id: "trace_1",
+      source_trace_id: "net_a", // points directly to source_net
```
Comment (MustafaMulla29) [inline tests/pcb/repros/repro19-segment-net-assignment-bug.test.tsx:32]: i dont think `source_trace_id` is assigned to something like `net_a`. See core, it alwasys assigns like `source_trace_...`

**Review by MustafaMulla29 — CHANGES_REQUESTED** [review]
See comment

File: tests/pcb/repros/repro19-segment-net-assignment-bug.test.tsx (hunk 2 above)
Comment (itisrohit) [reply tests/pcb/repros/repro19-segment-net-assignment-bug.test.tsx:32]: > i dont think `source_trace_id` is assigned to something like `net_a`. See core, it alwasys assigns like `source_trace_...`

Normally `@tscircuit/core` assigns it to a `source_trace_...` ID, but when using external autorouters (like `capacity-autorouter`), they route at the net-level and generate `pcb_trace` elements where `source_trace_id` is set directly to the `source_net_...` ID.

There are two existing files in this repository that prove this structure:

1. **Existing Code Proof**: In `lib/pcb/stages/AddViasStage.ts` (lines 160-169), we explicitly handle `source_trace_id` pointing to a `source_net` ID for vias:
   ```typescript
   // Bug 1 fix: source_trace_id may actually be a source_net ID
   // (capacity-autorouter sets it this way)
   if (!connectivityKey) {
     const sourceNet = this.ctx.db.source_net?.get(
       pcbTrace.source_trace_id,
     )
     if (sourceNet?.subcircuit_connectivity_map_key) {
       connectivityKey = sourceNet.subcircuit_connectivity_map_key
     }
   }

Comment (MustafaMulla29) [reply tests/pcb/repros/repro19-segment-net-assignment-bug.test.tsx:32]: Autorouter is ran always through core

Comment (itisrohit) [reply tests/pcb/repros/repro19-segment-net-assignment-bug.test.tsx:32]: > Autorouter is ran always through core

Normally `source_trace_id` is a trace ID, but external autorouters like `capacity-autorouter` sometimes output it as a `source_net` ID without a `connection_name`. 

This is the exact same structure we handle for vias in `AddViasStage.ts` (from PR #187):
`// Bug 1 fix: source_trace_id may actually be a source_net ID (capacity-autorouter sets it this way)`

This repro test is to verify we can handle that same edge case for trace segments in `AddTracesStage.ts`.
